src/firestore/firekit.ts

```diff
@@ -639,9 +641,18 @@ export class RoarFirekit {
     this._verifyAuthentication();
     const docRef = doc(this.dbRefs!.admin.assignments, administrationId);
     const docSnap = await getDoc(docRef);
-
-    if (docSnap.exists()) {
-      return docSnap.data() as IAssignmentData;
+    if(docSnap.exists()) {
+      const docData = docSnap.data() as IAssessmentData;
```

Comment:
I still think this should read
```
const docData = docSnap.data() as IAssignmentData;
```
and that this will remove the need for your `as unknown as IAssignmentData` type casting below.